backend/fms_core/template_importer/row_handlers/sample_submission/pool.py

```diff
@@ -80,9 +82,35 @@ def process_row_inner(self, samples_info, pool, seq_instrument_type, reception_d
                                                     "Index {2} for sample {3} are not different enough {4}.", [indices[i].name, samples_name[i], indices[j].name, samples_name[j], index_distance]))
                           self.warnings["index_collision"] = index_warnings
 
+            equal_volume_ratio = True
+            for sample in samples_info:
+                if sample.get("volume_ratio", None) is not None:
+                    equal_volume_ratio = False
+
+            exp = Decimal(f'1E-{DerivedBySample.VOLUME_RATIO_DECIMAL_PLACES}')
+            total_volume_ratio = 0
+            for sample in samples_info:
+                volume_ratio = sample.get("volume_ratio", None)
+                if not equal_volume_ratio:
+                    if volume_ratio is None:
+                        self.errors["volume_ratio"] = [f"All samples in the pool {pool['name']} must either define volume ratio or not at the same time."]
+                    else:
+                        total_volume_ratio += Decimal(volume_ratio)
+                else:
+                    self.warnings["volume_ratio"] = [f"Each sample in the pool {pool['name']} will have a volume ratio of {Decimal(1/len(samples_info)).quantize(exp)}."]
+                    total_volume_ratio = 1
+                    if volume_ratio:
```

Comment:
you are testing a condition that cannot happen unless you made a mistake in your code... This test and error should never be called... better remove them.